var/spack/repos/builtin/packages/hip/package.py

```diff
@@ -291,7 +291,13 @@ def get_paths(self):
             # We assume self.spec.prefix is /opt/rocm-x.y.z/hip and rocm has a
             # default installation with everything installed under
             # /opt/rocm-x.y.z
-            rocm_prefix = Prefix(os.path.dirname(self.spec.prefix))
+            # Note that since the key hip library can also exist at the top of the
+            # /opt/rocm-x.y.z/lib tree, it is possible that the package is detected
+            # without the correct prefix.  Work around it.
```

Comment:
This is confusing me.  If you've registered `hip` as an external but it's in `/opt/rocm-x.y.z/` instead of `/opt/rocm-x.y.z/hip`, shouldn't we make Spack get the external definition right?

I *think* the better way to do this is to handle this in an override of `determine_spec_details()` so that external detection succeeds, but I don't know if that handles all the cases if `libamdhip64` can really exist in `/opt/rocm-x.y.z/lib` and `rocm-x.y.z/hip/lib`.  If it is in the former will it also exist in the latter if the latter exists?

Reply:
@tgamblin I believe that it will exist in both of them, but part of the problem is that the modules which set this up are inconsistent in what they expose.  So if you use one it will expose the more complete path, and the other one will only expose it at the top.

Reply:
Can you say how you’re detecting this? Are you looking at variables from modules in a script?  Which ones?

Reply:
@tgamblin I just have it looking for both paths and if it doesn't find it under a `/hip/` directory then it asserts that it is at the top of the `rocm` tree.